src/Interpreters/Streaming/Aggregator.cpp

```diff
@@ -1026,28 +1134,12 @@ template <bool use_compiled_functions>
         }
     }
 
-    return should_finalize;
-}
+    if (hasExpandedData())
+        UpdatedDataEx::addBatchSinglePlace(row_begin, row_end, res, aggregate_instructions ? aggregate_instructions->delta_column : nullptr);
```

Comment:
if tracking updates is there, does it mean delta column shall aways be there otherwise there is something wrong during  query planning ?

Reply:
For `TrackingUpdatesWithRetract`, the delta_col may be nullptr.